tests/integration/test_compositions.py

```diff
@@ -58,13 +58,56 @@ def test_looping_embarrassingly_parallel_step(test_specific_results_dir: Path) -
         ).exists()
 
 
+EP_SECTION_MAPPING = {
+    "parallel_step": {
+        "pipeline_spec": "pipeline_parallel_step.yaml",
+        "schema_name": "ep_parallel_step",
+        "node_prefix": "parallel_split",
+    },
+    "loop_step": {
+        "pipeline_spec": "pipeline_loop_step.yaml",
+        "schema_name": "ep_loop_step",
+        "node_prefix": "loop",
+    },
+}
+
+
 @pytest.mark.slow
-def test_embarrassingly_parallel_parallel_step(test_specific_results_dir: Path) -> None:
-    pipeline_specification = EP_SPECIFICATIONS_DIR / "pipeline_parallel_step.yaml"
+@pytest.mark.parametrize(
+    "step_type, expected_counter, num_rows_multiplier",
+    [
+        (
+            "loop_step",
+            2,
```

Comment:
Asside from stringy stuff encapsulated in the mapping above, this is the fundamental difference in outputs between an embarrassingly parallel LoopStep (this PR, specifically) and ParallelStep (already merged): When looping, we use the container twice (once per loop) and so the counter is 2. Further, nothing about looping expands the length of the data. Meanwhile, when branching in parallel, each branch hits the container only once and so the counter should be 1. Then, we aggregate all resulting files which means 2x in length (b/c 2x parallel branches)